### PR TITLE
remove arc.initialize method, replace by set/getContractInfos

### PR DESCRIPTION
**Pull request by jellegerbrandy**
- "fixes" sort of 265 (by rendering explicit what it does)

### CODE REVIEW

File: test/arc.spec.ts [hunk 1]
```diff
@@ -38,10 +39,22 @@ describe('Arc ', () => {
     })
     expect(arc).toBeInstanceOf(Arc)
 
-    expect(() => arc.sendQuery(`{daos {id}}`)).toThrowError(/no connection/i)
+    expect(() => arc.sendQuery(gql`{daos {id}}`)).toThrowError(/no connection/i)
   })
 
-  it('arc.getContractAddresses() should work', async () => {
+  it('Arc is usable without knowning about contrats', () => {
```
Comment (tibetsprague) [inline test/arc.spec.ts]: "knowing" and "contracts"

**Comment by jellegerbrandy** [comment]
The point of this PR is to remove the `initialize()` function, because the name was confusing. It did not really "initiaize" arc - instead, what it was doing is to provide the information about contracts to arc. This led to confusion in alchemy. So it seems better to do that explicitly.

About a "reset" funtion:  I am not sure what that would do, or even what the function signature should look like (`arc.reset(newOptions)` ?). If you want to change the settings (say use another web3 provider), you can simply create a new Arc instance and use that. But perhaps I am missing a point?

**Comment by tibetsprague** [comment]
ok I guess, just seems weird for an external user of this library to have to call getContractInfo and then setContractInfo, seems like a single call to arc should just do that, maybe call it loadContractInfo

**Comment by jellegerbrandy** [comment]
I see your point. But the idea is that you could get the contract info also from sources other than the subgraph, like the migration.json file that we used before.

**Pull request merged**